src/app/data-explorer/selection-compact/selection-compact.component.ts

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022-2023 University of Oxford
+and Health and Social Care Information Centre, also known as NHS Digital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import { Component, Input } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { DataAccessRequestsSourceTargetIntersections } from '../data-requests.service';
+import {
+  SelectionExpandedDialogComponent,
+  SelectionExpandedDialogData,
+} from '../selection-expanded-dialog/selection-expanded-dialog.component';
+import { SelectionService } from '../selection.service';
+
+@Component({
+  selector: 'mdm-selection-compact',
+  templateUrl: './selection-compact.component.html',
+  styleUrls: ['./selection-compact.component.scss'],
+})
+export class SelectionCompactComponent {
+  @Input() sourceTargetIntersections: DataAccessRequestsSourceTargetIntersections;
+  constructor(private selectionService: SelectionService, private matDialog: MatDialog) {
+    this.sourceTargetIntersections = {
+      dataAccessRequests: [],
+      sourceTargetIntersections: [],
+    };
+  }
+
+  get selectedElements$() {
+    return this.selectionService.list$;
+  }
+
+  clearSelection() {
+    this.selectionService.clearSelection();
+  }
+
+  showSelection() {
+    if (this.sourceTargetIntersections !== undefined) {
+      this.matDialog.open<SelectionExpandedDialogComponent, SelectionExpandedDialogData>(
```

Comment:
I've found a bug which seems to happen when I open this dialog, but I can't see an obvious cause for it at the moment.

1. Search for any term which will take you to the search listing page - note the URL will include query params like "?search=batch` etc
2. Tick a search result to add to the selection and make the "Current selection" box appear on the left.
3. Click on the "X element(s) selected" link

The `SelectionExpandedDialogComponent` opens correctly, but notice behind the dialog that the search listing page refreshes. It has somehow triggered a new search with no query params, effectively doing a search over every element in the catalogue.

I'm not sure why this is happening, please could you investigate.

Reply:
I've re-tested and the odd refresh behaviour has now gone 👍